### PR TITLE
Fix for GVRCameraSceneObject

### PR DESCRIPTION
**Pull request by rahul27**
Register the draw frame listener only after acquiring the
surface texture.

GearVRf-DCO-1.0-Signed-off-by: Rahul Rudradevan
r.rudradevan@samsung.com

### CODE REVIEW

**Comment by rahul27** [comment]
Both this and PR https://github.com/gearvrf/GearVRf-Demos/pull/398 fix the gvr-sceneobjects sample for daydream.

**Comment by liaxim** [comment]
Good fix. Will merge after lunch.

**Comment by rahul27** [comment]
Will change this pull request to fix the scene object to handle state changes.

**Comment by rahul27** [comment]
This PR and the sample are ready for review.

File: GVRf/Framework/framework/src/main/java/org/gearvrf/scene_objects/GVRCameraSceneObject.java [hunk 1]
```diff
@@ -81,29 +91,166 @@ public GVRCameraSceneObject(GVRContext gvrContext, GVRMesh mesh,
...
+            return;
+        }
+
+        camera.stopPreview();
+        camera.release();
+        camera = null;
+        isCameraOpen = false;
+    }
+
+    private IActivityEvents activityEvents = new IActivityEvents() {
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/scene_objects/GVRCameraSceneObject.java]: Fyi, there is empty implementation in GVREventListeners.ActivityEvents that can be used to eliminate the noop methods.

**Comment by liaxim** [comment]
Looks good. Will merge Wednesday after lunch.

File: GVRf/Framework/framework/src/main/java/org/gearvrf/scene_objects/GVRCameraSceneObject.java (hunk 1 above)
Comment (rahul27) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/scene_objects/GVRCameraSceneObject.java]: Let me add that in.

**Comment by rahul27** [comment]
Done.

**Pull request merged**